extensions/schema-compare/src/dialogs/schemaCompareDialog.ts

```diff
@@ -874,13 +874,13 @@ export class SchemaCompareDialog {
 				finalName = c.options.connectionName;
 			}
 
-			// use previously selected server or current connection if there is one
-			if (endpointInfo && !isNullOrUndefined(endpointInfo.serverName) && !isNullOrUndefined(endpointInfo.serverDisplayName)
-				&& c.options.server.toLowerCase() === endpointInfo.serverName.toLowerCase()
-				&& finalName.toLowerCase() === endpointInfo.serverDisplayName.toLowerCase()) {
+			// use current connection else use previously selected server if there is one
+			if (c.connectionId === this.connectionId) {
 				idx = count;
-			}
-			else if (c.connectionId === this.connectionId) {
+			} else if (endpointInfo && !isNullOrUndefined(endpointInfo.serverName) && !isNullOrUndefined(endpointInfo.serverDisplayName)
```

Comment:
nit: it looks like both the `if` and `else` blocks have the same code.  I think you could combine into a single condition with an `||` operator.